File: 162.find-peak-element.go
func findPeakElement(nums []int) int {
    return find(nums, 0, len(nums)-1)
}

func find(nums []int, lo, hi int) int {
    if lo == hi {
        return lo
    }
    mid := (lo + hi) / 2
    if nums[mid] > nums[mid+1] {
        return find(nums, lo, mid)
    }
    return find(nums, mid+1, hi)
}

File: 162.find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int i = 0, j = nums.length - 1;
        while (i < j) {
            int mid = (int)Math.floor((i+j)/2);
            if (nums[mid] < nums[mid+1]){
                i = mid+1;
            } else {
                j = mid;
            }
        }
        return j;
    }
}
